crates/agglayer-grpc-api/src/network_state_service.rs

```diff
@@ -0,0 +1,38 @@
+use agglayer_grpc_server::node::v1::network_state_service_server::NetworkStateService;
+use agglayer_grpc_types::node::v1::{
+    GetLatestKnownCertificateHeaderRequest, GetLatestKnownCertificateHeaderResponse,
+    GetLatestPendingCertificateHeaderRequest, GetLatestPendingCertificateHeaderResponse,
+    GetLatestSettledCertificateHeaderRequest, GetLatestSettledCertificateHeaderResponse,
+};
+
+use crate::Server;
+
+#[tonic::async_trait]
+impl NetworkStateService for Server {
+    async fn get_latest_known_certificate_header(
+        &self,
+        request: tonic::Request<GetLatestKnownCertificateHeaderRequest>,
+    ) -> Result<tonic::Response<GetLatestKnownCertificateHeaderResponse>, tonic::Status> {
+        let _request = request.into_inner();
```

Comment:
Not to repeat further, add todo with explanation to all unfinished services.
Easier to search for/filter/understand